### ISSUE TITLE
Add linter for superfluous primitive coercion calls

### ISSUE THREAD

**Issue by hugoduncan** [clj-kondo/clj-kondo#2709]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Problem Description**

Redundant primitive coercion calls add noise to code and may indicate a misunderstanding of return types. For example:

```clojure
(defn calculate ^double [x] 
  (* x 2.0))

;; This coercion is unnecessary - calculate already returns double
(double (calculate 5))
```

**Proposed Solution**

A new linter (e.g., `:redundant-primitive-coercion`) that warns when a primitive coercion function (`double`, `float`, `long`, `int`, `short`, `byte`, `char`, `boolean`) is applied to an expression that already returns that primitive type.

Example warning:
```
src/foo.clj:10:1: warning: Redundant call to double - expression already returns double
```

This should detect:
- Functions with primitive return type hints: `(defn f ^double [x] ...)`
- Java interop returning primitives
- Other primitive coercion calls: `(double (double x))`

**Alternatives Considered**

Could be implemented as a hook, but a built-in linter would benefit all users without additional configuration.

**Additional Context**

This is similar in spirit to existing redundancy linters like `:redundant-str-call`.
